### PR TITLE
Correlations: Add GetCorrelation(s) HTTP APIs

### LINKED ISSUE #52582: Glue Correlations: GetCorrelations API endpoints
As part of [#51323](https://github.com/grafana/grafana/issues/51323) we need APIs endpoint to retrieve existing correlations.

## Get a single correlation

**Path**: `/datasources/uid/{uid}/correlations/{correlationUid}`
**Method**: `GET`

Gets a single correlation

## Get all correlations originating from a given data source

**Path**: `/datasources/uid/{uid}/correlations`
**Method**: `GET`

Gets all the correlations originating from the datasource specified in the path

## Get all correlations

**Path**: `/datasources/correlations`
**Method**: `GET`

Gets all the correlations.

## Other notes
also add indices for `uid` and `source_uid`  column in `correlations` table

### PR DESCRIPTION
**Pull request by Elfo404**
Fixes #52582 

- [x] Get single correlation (`GET /api/datasources/uid/{dsUID}/correlations/{uid}`) b6677556fb03c4de61d81b6c5fa02dd9ab6f527d
- [x] Get all correlations for a given DS (`GET /api/datasources/uid/{dsUID}/correlations/`) 2202ce02058b53af11fe154cdf07529e531b71ad
- [x] Get all correlations (`GET /api/datasources/correlations`) 0dea6eb997d3f6e2d9bfbc604785f51e6438e82e

TODO:
- [x] add tests 235087ffcd93ff66045e091d1a5f387e1a0064f0
- [x] change casing (as per https://github.com/grafana/grafana/pull/51801#discussion_r930165948)
- [x] add indices to db for `uid` and `source_uid` 19c0751d928391617c8850e8823206e921184136

### CODE REVIEW

File: pkg/tests/api/correlations/correlations_read_test.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
+package correlations
```
Comment (Elfo404) [inline pkg/tests/api/correlations/correlations_read_test.go:1]: there are 3 skipped tests in this file, all related to what i think is some accesscontrol quirk and the way the test env is set up as viewer or editors do not get `datasource:read` permission.

Comment (Elfo404) [reply pkg/tests/api/correlations/correlations_read_test.go:1]: yep, GE tests fail as expected.

**Review by chri2547 — COMMENTED** [review]
Doc changes look fine. Please merge after code reviewed and approved.

File: pkg/tests/api/correlations/correlations_read_test.go (hunk 1 above)
Comment (Elfo404) [reply pkg/tests/api/correlations/correlations_read_test.go:1]: fixed.

**Comment by Elfo404** [comment]
@grafana/backend-platform any chance i can get a review on this?

**Review by gelicia — APPROVED** [review]

**Pull request merged**